### PR TITLE
Add intersphinx_registry and toml-sort

### LINKED ISSUE #1300: Use intersphinx_registry
in our `conf.py`, we should use https://github.com/Quansight-Labs/intersphinx_registry and we should add mne-bids to that registry.

### PR DESCRIPTION
**Pull request by mscheltienne**
Add `toml-sort` and `intersphinx_registry` to the tooling chain.
Closes #1300

### CODE REVIEW

**Review by larsoner — APPROVED** [review]

**Review by drammock — COMMENTED** [review]
I suggest adding the following to `pyproject.toml`:

```toml
[tool.tomlsort]
sort_table_keys = true
sort_inline_tables = true
sort_inline_arrays = true
spaces_before_inline_comment = 2
spaces_indent_inline_array = 4
trailing_comma_inline_array = true
```
the `sort_*` settings will make order fully deterministic, and the other settings will make the result more similar to how `ruff` formats python code, which IMO makes manual edits slightly easier (e.g., hitting TAB in the IDE will yield the right number of spaces by default)

**Review by sappelhoff — APPROVED** [review]

**Comment by mscheltienne** [comment]
I was certain I putted:

```
[tool.tomlsort]
all = true
ignore_case = true
trailing_comma_inline_array = true
```

But now I can't find it..
This config is also deterministic and IIRC `all` replaces your 3 `sort` lines. For the spacing, I would argue it's a configuration of the IDE. If I hit TAB in a `pyproject.toml`, I get a spacing of 2, not 4.

File: pyproject.toml [hunk 1]
```diff
@@ -87,24 +82,64 @@ dev = ["mne_bids[test,doc,full]", "pre-commit"]
...
+  # numba with NumPy dev
+  "ignore:`np.MachAr` is deprecated.*:DeprecationWarning",
+  # old MNE _fake_click
+  "ignore:The .*_event function was deprecated in Matplotlib.*:",
+  "ignore:datetime\\.datetime\\.utcfromtimestamp.* is deprecated and scheduled for removal in a future version.*:DeprecationWarning",
+  # matplotlib
+  "ignore:Figure.*is non-interactive.*cannot be shown:UserWarning",
+  # NumPy 2.1 bug (probably)
+  "ignore:__array__ implementation doesn.*:DeprecationWarning"
+]
```
Comment (mscheltienne) [inline pyproject.toml:142]:
```suggestion
  "ignore:__array__ implementation doesn.*:DeprecationWarning"
]

[tool.tomlsort]
all = true
ignore_case = true
trailing_comma_inline_array = true

```

**Comment by drammock** [comment]
> This config is also deterministic and IIRC `all` replaces your 3 `sort` lines. For the spacing, I would argue it's a configuration of the IDE. If I hit TAB in VSCode in a `pyproject.toml`, I get a spacing of 2, not 4.

yep, confirmed that `all` is equiv to the 3 sort lines I suggested. (I was skimming docs quickly and assumed it meant "format all discovered files" or something).

Your point about spacing is true --- users *can* configure their IDEs to do tabs differently for different file formats.  But if we use the setting I suggested, they won't *need* to do that... though I guess it's possible that my suggested setting will lead folks to need to undo their 2-space tab settings?

For other settings though (e.g., 2 spaces before inline comments) it's less about IDEs and more about muscle memory: chances are if I'm typing an inline comment I'll use 2 spaces, and when reading code I'm used to things being looking a certain way (I'm guessing that's why you added trailing comma --- after months/years of using black/ruff, it's starting to look "wrong" if it's not there?)

If you're not convinced that's OK. I don't insist on anything other than the sorting, which you've already done.

**Comment by mscheltienne** [comment]
> For other settings though (e.g., 2 spaces before inline comments) it's less about IDEs and more about muscle memory: chances are if I'm typing an inline comment I'll use 2 spaces, and when reading code I'm used to things being looking a certain way (I'm guessing that's why you added trailing comma --- after months/years of using black/ruff, it's starting to look "wrong" if it's not there?)

I completely agree and wil ladd the 2 spaces before inline comments at least ;)

> Your point about spacing is true --- users can configure their IDEs to do tabs differently for different file formats. But if we use the setting I suggested, they won't need to do that... though I guess it's possible that my suggested setting will lead folks to need to undo their 2-space tab settings?

My issue here, is that TBH, I don't think I actually configured anything here and yet that's the default I got (it might be from the `Even Better TOML` externsion that I have installed). So in this case, we have 2 different 'default' behavior, which is not great. I'm enclined to keep the default from `toml-sort` for this one.

File: pyproject.toml [hunk 2]
```diff
@@ -1,155 +1,155 @@
...
-    "ignore:The .*_event function was deprecated in Matplotlib.*:",
-    "ignore:datetime\\.datetime\\.utcfromtimestamp.* is deprecated and scheduled for removal in a future version.*:DeprecationWarning",
-    # matplotlib
-    "ignore:Figure.*is non-interactive.*cannot be shown:UserWarning",
-    # NumPy 2.1 bug (probably)
-    "ignore:__array__ implementation doesn.*:DeprecationWarning",
-]
+[tool.tomlsort]
+all = true
+ignore_case = true
```
Comment (mscheltienne) [inline pyproject.toml:154]:
```suggestion
ignore_case = true
spaces_before_inline_comment = 2
```

**Review by drammock — APPROVED** [review]

**Pull request merged**